From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alpha <alphakr93@outlook.com>
Date: Wed, 19 Jan 2022 23:05:37 +0900
Subject: [PATCH] Prismarine Config Files


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index cec93a6833acf3a4c5a55b4183bd3d45ef3e3362..7e1986f10b5dab0796e8a87c16f96210b6e7667b 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1794,6 +1794,7 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         // Purpur end
 
         // Prismarine start
+        @NotNull
         public org.bukkit.configuration.file.YamlConfiguration getPrismarineConfig() {
             throw new UnsupportedOperationException("Not supported yet.");
         }
