From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alpha <alphakr93@outlook.com>
Date: Wed, 19 Jan 2022 23:18:02 +0900
Subject: [PATCH] Do not send useless entity packets


diff --git a/src/main/java/net/minecraft/server/level/ServerEntity.java b/src/main/java/net/minecraft/server/level/ServerEntity.java
index e3d7a80f8c4b68933875ef90006b79776b4ab000..2cd09fea929577c9a52205fe8e53532e65292604 100644
--- a/src/main/java/net/minecraft/server/level/ServerEntity.java
+++ b/src/main/java/net/minecraft/server/level/ServerEntity.java
@@ -189,6 +189,7 @@ public class ServerEntity {
                         this.teleportDelay = 0;
                         packet1 = new ClientboundTeleportEntityPacket(this.entity);
                     }
+                    if (net.prismarineteam.prismarine.PrismarineConfig.doNotSendUselessEntityPackets && isUselessEntityPacket(packet1)) packet1 = null; // Prismarine
                 }
 
                 if ((this.trackDelta || this.entity.hasImpulse || this.entity instanceof LivingEntity && ((LivingEntity) this.entity).isFallFlying()) && this.tickCount > 0) {
@@ -257,6 +258,22 @@ public class ServerEntity {
 
     }
 
+    // Prismarine start
+    private boolean isUselessEntityPacket(Packet<?> possibleUselessEntityPacket) {
+        if (possibleUselessEntityPacket instanceof ClientboundMoveEntityPacket) {
+            ClientboundMoveEntityPacket p = (ClientboundMoveEntityPacket) possibleUselessEntityPacket;
+            if (possibleUselessEntityPacket instanceof ClientboundMoveEntityPacket.Pos) {
+                return p.getXa() == 0 && p.getYa() == 0 && p.getZa() == 0;
+            } else if (possibleUselessEntityPacket instanceof ClientboundMoveEntityPacket.PosRot) {
+                return p.getXa() == 0 && p.getYa() == 0 && p.getZa() == 0 && p.getxRot() == 0 && p.getyRot() == 0;
+            } else if (possibleUselessEntityPacket instanceof ClientboundMoveEntityPacket.Rot) {
+                return p.getxRot() == 0 && p.getyRot() == 0;
+            }
+        }
+        return false;
+    }
+    // Prismarine end
+
     public void removePairing(ServerPlayer player) {
         this.entity.stopSeenByPlayer(player);
         player.connection.send(new ClientboundRemoveEntitiesPacket(new int[]{this.entity.getId()}));
diff --git a/src/main/java/net/prismarineteam/prismarine/PrismarineConfig.java b/src/main/java/net/prismarineteam/prismarine/PrismarineConfig.java
index 40e5718ed69146cbf1aef7819bda1b8822e2499d..29f18e26258c2d6b23bc8e601167da7dbe13f34b 100644
--- a/src/main/java/net/prismarineteam/prismarine/PrismarineConfig.java
+++ b/src/main/java/net/prismarineteam/prismarine/PrismarineConfig.java
@@ -145,4 +145,9 @@ public class PrismarineConfig {
         }
         return builder.build();
     }
+
+    public static boolean doNotSendUselessEntityPackets = true;
+    private static void doNotSendUselessEntityPackets() {
+        doNotSendUselessEntityPackets = getBoolean("settings.do-not-send-useless-entity-packets", doNotSendUselessEntityPackets);
+    }
 }
\ No newline at end of file
