From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alpha <alphakr93@outlook.com>
Date: Sun, 9 Jan 2022 11:21:45 +0900
Subject: [PATCH] Change Pufferfish main command like vanilla


diff --git a/src/main/java/gg/pufferfish/pufferfish/PufferfishCommand.java b/src/main/java/gg/pufferfish/pufferfish/PufferfishCommand.java
index 020368da69b9a492155f6de6297f74732f4ab6ea..9c55b4fa92a5825082454c588d52ef6950ca2a1d 100644
--- a/src/main/java/gg/pufferfish/pufferfish/PufferfishCommand.java
+++ b/src/main/java/gg/pufferfish/pufferfish/PufferfishCommand.java
@@ -40,14 +40,17 @@ public class PufferfishCommand extends Command {
     @Override
     public boolean execute(CommandSender sender, String commandLabel, String[] args) {
         if (!testPermission(sender)) return true;
-        String prefix = ChatColor.of("#12fff6") + "" + ChatColor.BOLD + "Pufferfish » " + ChatColor.of("#e8f9f9");
 
         if (args.length != 1) {
-            sender.sendMessage(prefix + "Usage: " + usageMessage);
+            sender.sendMessage(ChatColor.RED + "Usage: " + usageMessage); // Prismarine
             args = new String[]{"version"};
         }
 
         if (args[0].equalsIgnoreCase("reload")) {
+            // Prismarine start
+            Command.broadcastCommandMessage(sender, org.bukkit.ChatColor.RED + "Please note that this command is not supported and may cause issues.");
+            Command.broadcastCommandMessage(sender, org.bukkit.ChatColor.RED + "If you encounter any issues please use the /stop command to restart your server.");
+            // Prismarine end
             MinecraftServer console = MinecraftServer.getServer();
             try {
                 PufferfishConfig.load();
@@ -58,9 +61,14 @@ public class PufferfishCommand extends Command {
             }
             console.server.reloadCount++;
 
-            Command.broadcastCommandMessage(sender, prefix + "Pufferfish configuration has been reloaded.");
+            // Prismarine start
+            Command.broadcastCommandMessage(sender, org.bukkit.ChatColor.GREEN + "Pufferfish config reload complete.");
         } else if (args[0].equalsIgnoreCase("version")) {
-            Command.broadcastCommandMessage(sender, prefix + "This server is running " + Bukkit.getName() + " version " + Bukkit.getVersion() + " (Implementing API version " + Bukkit.getBukkitVersion() + ")");
+            Command verCmd = org.bukkit.Bukkit.getServer().getCommandMap().getCommand("version");
+            if (verCmd != null) {
+                return verCmd.execute(sender, commandLabel, new String[0]);
+            }
+            // Prismarine end
         }
 
         return true;
