From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alpha <alphakr93@outlook.com>
Date: Thu, 20 Jan 2022 10:03:50 +0900
Subject: [PATCH] Use normal context, which is already lazy


diff --git a/src/main/java/io/papermc/paper/util/CollisionUtil.java b/src/main/java/io/papermc/paper/util/CollisionUtil.java
index 58629451977c89db2fa895bde946135784a0d8bc..359f6f01d1103de60d5ff7de82a1bce88165d250 100644
--- a/src/main/java/io/papermc/paper/util/CollisionUtil.java
+++ b/src/main/java/io/papermc/paper/util/CollisionUtil.java
@@ -519,7 +519,11 @@ public final class CollisionUtil {
                             if ((edgeCount != 1 || blockData.shapeExceedsCube()) && (edgeCount != 2 || blockData.getBlock() == Blocks.MOVING_PISTON)) {
                                 mutablePos.set(blockX, blockY, blockZ);
                                 if (collisionShape == null) {
-                                    collisionShape = new LazyEntityCollisionContext(entity);
+                                    if (net.prismarineteam.prismarine.PrismarineConfig.useNormalCollisionContext) {
+                                        collisionShape = CollisionContext.of(entity);
+                                    } else {
+                                        collisionShape = new LazyEntityCollisionContext(entity);
+                                    }
                                 }
                                 VoxelShape voxelshape2 = blockData.getCollisionShape(getter, mutablePos, collisionShape);
                                 if (voxelshape2 != Shapes.empty()) {
diff --git a/src/main/java/net/prismarineteam/prismarine/PrismarineConfig.java b/src/main/java/net/prismarineteam/prismarine/PrismarineConfig.java
index 29f18e26258c2d6b23bc8e601167da7dbe13f34b..4dad7fc69bd213fcd20c24b032e1fa70cc6d0366 100644
--- a/src/main/java/net/prismarineteam/prismarine/PrismarineConfig.java
+++ b/src/main/java/net/prismarineteam/prismarine/PrismarineConfig.java
@@ -150,4 +150,9 @@ public class PrismarineConfig {
     private static void doNotSendUselessEntityPackets() {
         doNotSendUselessEntityPackets = getBoolean("settings.do-not-send-useless-entity-packets", doNotSendUselessEntityPackets);
     }
+
+    public static boolean useNormalCollisionContext = true;
+    private static void useAirplaneCollisionContext() {
+        useNormalCollisionContext = getBoolean("settings.use-normal-collision-context", useNormalCollisionContext);
+    }
 }
\ No newline at end of file
