From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alpha <alphakr93@outlook.com>
Date: Sat, 8 Jan 2022 23:02:19 +0900
Subject: [PATCH] Add back "Lobotomize stuck villagers" Purpur Patch


diff --git a/src/main/java/net/minecraft/world/entity/npc/Villager.java b/src/main/java/net/minecraft/world/entity/npc/Villager.java
index 9755517f0c1d66db5bfd00946114fab5db6776d6..b5114a0bbf83c8dd7dbcf1326ce6e235b510e715 100644
--- a/src/main/java/net/minecraft/world/entity/npc/Villager.java
+++ b/src/main/java/net/minecraft/world/entity/npc/Villager.java
@@ -138,6 +138,7 @@ public class Villager extends AbstractVillager implements ReputationEventHandler
         return villageplacetype == PoiType.MEETING;
     });
     private final int brainTickOffset; // Purpur
+    private boolean lobotomized = false; // Prismarine
 
     public long nextGolemPanic = -1; // Pufferfish
 
@@ -187,6 +188,24 @@ public class Villager extends AbstractVillager implements ReputationEventHandler
         return level.purpurConfig.villagerCanBeLeashed && !this.isLeashed();
     }
 
+    // Prismarine start
+    private boolean canTravelTo(BlockPos pos) {
+        net.minecraft.world.level.pathfinder.Path to = navigation.createPath(pos, 0);
+        return to != null && to.nodes.size() > 0;
+    }
+
+    private boolean canTravelFrom(BlockPos pos) {
+        return canTravelTo(pos.north()) || canTravelTo(pos.east()) || canTravelTo(pos.south()) || canTravelTo(pos.west());
+    }
+
+    private boolean isLobotomized() {
+        if ((level.getGameTime() + brainTickOffset) % level.prismarineConfig.villagerLobotomizeCheck == 0) {
+            this.lobotomized = !canTravelFrom(blockPosition().above());
+        }
+        return this.lobotomized;
+    }
+    // Prismarine end
+
     @Override
     public boolean isSensitiveToWater() {
         return this.level.purpurConfig.villagerTakeDamageFromWater;
diff --git a/src/main/java/net/prismarineteam/prismarine/PrismarineWorldConfig.java b/src/main/java/net/prismarineteam/prismarine/PrismarineWorldConfig.java
index cf494171713b5fc12e9627774f2705b911b54bd2..e2f0f17e4a53feabbeb5ac71d97e6b5c21da24df 100644
--- a/src/main/java/net/prismarineteam/prismarine/PrismarineWorldConfig.java
+++ b/src/main/java/net/prismarineteam/prismarine/PrismarineWorldConfig.java
@@ -74,4 +74,7 @@ public class PrismarineWorldConfig {
         final Map<String, Object> value = PrismarineConfig.getMap("world-settings." + worldName + "." + path, null);
         return value.isEmpty() ? fallback : value;
     }
+
+    public boolean villagerLobotomizeEnabled = getBoolean("mobs.villager.lobotomize.enabled", false);
+    public int villagerLobotomizeCheck = getInt("mobs.villager.lobotomize.check-interval", 60);
 }
\ No newline at end of file
