From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alpha <alphakr93@outlook.com>
Date: Mon, 17 Jan 2022 11:52:13 +0900
Subject: [PATCH] Add back "Use normal context, which is already lazy" Airplane
 patch until Paper 1.18 Perfomance patch


diff --git a/src/main/java/io/papermc/paper/util/CollisionUtil.java b/src/main/java/io/papermc/paper/util/CollisionUtil.java
index 58629451977c89db2fa895bde946135784a0d8bc..ac01cdfba1850240b88a78b660302544f5d0cda1 100644
--- a/src/main/java/io/papermc/paper/util/CollisionUtil.java
+++ b/src/main/java/io/papermc/paper/util/CollisionUtil.java
@@ -519,7 +519,8 @@ public final class CollisionUtil {
                             if ((edgeCount != 1 || blockData.shapeExceedsCube()) && (edgeCount != 2 || blockData.getBlock() == Blocks.MOVING_PISTON)) {
                                 mutablePos.set(blockX, blockY, blockZ);
                                 if (collisionShape == null) {
-                                    collisionShape = new LazyEntityCollisionContext(entity);
+                                    // collisionShape = new LazyEntityCollisionContext(entity); // Airplane // Prismarine - Until 1.18 Performance Patch
+                                    collisionShape = CollisionContext.of(entity);
                                 }
                                 VoxelShape voxelshape2 = blockData.getCollisionShape(getter, mutablePos, collisionShape);
                                 if (voxelshape2 != Shapes.empty()) {
@@ -599,6 +600,9 @@ public final class CollisionUtil {
         }
     }
 
+    // Prismarine start - Until 1.18 Performance Patch
+    // Airplane start - unneeded
+    /*
     public static final class LazyEntityCollisionContext extends EntityCollisionContext {
 
         private CollisionContext delegate;
@@ -632,6 +636,9 @@ public final class CollisionUtil {
             return this.getDelegate().canStandOnFluid(state, fluid);
         }
     }
+     */
+    // Airplane end
+    // Prismarine end
 
     private CollisionUtil() {
         throw new RuntimeException();
